test/gw/prior_test.py

```diff
@@ -170,6 +170,44 @@ def test_add_constraint_prior_not_redundant(self):
         )
         self.assertFalse(self.bbh_prior_dict.test_has_redundant_keys())
 
+    def test_is_cosmological_true(self):
+        self.bbh_prior_dict["luminosity_distance"] = bilby.gw.prior.UniformComovingVolume(
+            minimum=10, maximum=10000, name="luminosity_distance"
+        )
+        self.assertTrue(self.bbh_prior_dict.is_cosmological)
+
+    def test_is_cosmological_false(self):
+        del self.bbh_prior_dict["luminosity_distance"]
+        self.assertFalse(self.bbh_prior_dict.is_cosmological)
+
+    def test_check_valid_cosmology(self):
+        self.bbh_prior_dict["luminosity_distance"] = bilby.gw.prior.UniformComovingVolume(
+            minimum=10, maximum=10000, name="luminosity_distance"
+        )
+        self.assertTrue(self.bbh_prior_dict.check_valid_cosmology())
+
+    def test_check_valid_cosmology_raises_error(self):
+        self.bbh_prior_dict["luminosity_distance"] = bilby.gw.prior.UniformComovingVolume(
+            minimum=10, maximum=10000, name="luminosity_distance", cosmology="Planck15",
+        )
+        self.bbh_prior_dict["redshift"] = bilby.gw.prior.UniformComovingVolume(
+            minimum=0.1, maximum=1, name="redshift", cosmology="Planck15_LAL",
+        )
+        self.assertEqual(
+            self.bbh_prior_dict._cosmological_priors,
+            ["luminosity_distance", "redshift"],
+        )
+        self.assertRaises(ValueError, self.bbh_prior_dict.check_valid_cosmology)
+
+    def test_cosmology(self):
+        self.bbh_prior_dict["luminosity_distance"] = bilby.gw.prior.UniformComovingVolume(
+            minimum=10, maximum=10000, name="luminosity_distance"
+        )
+        self.assertEqual(
+            self.bbh_prior_dict.cosmology,
```

Comment:
Is there a good reason not to use `cosmology_equal` from `astropy` here?

Reply:
Good point, I don't think there is. I'll update it.

Reply:
Actually, on second thought, using `assertEqual` gives a more useful error message. `assertTrue` just gives `AssertionError: False is not true`

Reply:
Actually, seems like you can pass a custom message to `assertTrue`, for instance: `self.assertTrue(cosmology_equal(cosmo_1, cosmo_2), "The two cosmologies are not equivalent")`. Your call if you want to put this in though :)

Reply:
I think I'll leave it as is, the current version prints the cosmologies, meaning one can compare the values, which I think is useful.